### PR TITLE
IGNITE-13599 Switch tests should have better precision

### PR DESCRIPTION
**Pull request by anton-vinogradov**
Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [ ] There is a single JIRA ticket related to the pull request. 
- [ ] The web-link to the pull request is attached to the JIRA ticket.
- [ ] The JIRA ticket has the _Patch Available_ state.
- [ ] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [ ] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [ ] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [ ] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

File: modules/ducktests/src/main/java/org/apache/ignite/internal/ducktest/tests/cellular_affinity_test/CellularTxStreamer.java [hunk 1]
```diff
@@ -88,22 +97,37 @@
...
                 }
             }
 
             if (cnt % 1000 == 0)
-                log.info("APPLICATION_STREAMED " + cnt + " transactions [worst_latency=" + Arrays.toString(max) + "]");
+                log.info("APPLICATION_STREAMED " + cnt + " transactions [worst_latency=" + Arrays.toString(latencies) + "]");
+        }
+
+        List<String> result = new ArrayList<>();
+        Calendar calendar = Calendar.getInstance();
```
Comment (ivandasch) [inline modules/ducktests/src/main/java/org/apache/ignite/internal/ducktest/tests/cellular_affinity_test/CellularTxStreamer.java:118]: Calendar? ORLY? Have you considered LocalDateTime?

File: modules/ducktests/src/main/java/org/apache/ignite/internal/ducktest/tests/cellular_affinity_test/CellularTxStreamer.java [hunk 2]
```diff
@@ -88,22 +97,37 @@
...
+
+        for (int i = 0; i < precision; i++) {
+            calendar.setTimeInMillis(opStartTimes[i]);
+
+            result.add(latencies[i] + "ms started "
+                + calendar.get(Calendar.HOUR_OF_DAY) + ":"
```
Comment (ivandasch) [inline modules/ducktests/src/main/java/org/apache/ignite/internal/ducktest/tests/cellular_affinity_test/CellularTxStreamer.java:124]: Eeww, what about LocalDateTimeFormatter?

File: modules/ducktests/src/main/java/org/apache/ignite/internal/ducktest/tests/pme_free_switch_test/SingleKeyTxStreamerApplication.java [hunk 3]
```diff
@@ -46,11 +43,9 @@
 
             long start = System.currentTimeMillis();
 
-            cache.put(key++, key);
-
-            long finish = System.currentTimeMillis();
+            cache.put(key++ % 100, key); // Cycled update.
 
-            long time = finish - start;
+            long latency = System.currentTimeMillis() - start;
```
Comment (ivandasch) [inline modules/ducktests/src/main/java/org/apache/ignite/internal/ducktest/tests/pme_free_switch_test/SingleKeyTxStreamerApplication.java:48]: I don't hesitate to insist, that System.nanoTime() is the only option.

File: modules/ducktests/src/main/java/org/apache/ignite/internal/ducktest/tests/cellular_affinity_test/CellularTxStreamer.java [hunk 4]
```diff
@@ -58,26 +63,30 @@
...
-            key++;
+        List<Integer> cellKeys = new ArrayList<>();
+
+        int candidate = 0;
 
-            Collection<ClusterNode> nodes = aff.mapKeyToPrimaryAndBackups(key);
+        while (cellKeys.size() < 100) {
+            Collection<ClusterNode> nodes = aff.mapKeyToPrimaryAndBackups(++candidate);
 
             Map<Object, Long> stat = nodes.stream().collect(
```
Comment (ivandasch) [inline modules/ducktests/src/main/java/org/apache/ignite/internal/ducktest/tests/cellular_affinity_test/CellularTxStreamer.java:73]: Are you sure that creating the same map on each iteration is a good idea?

Comment (anton-vinogradov) [reply modules/ducktests/src/main/java/org/apache/ignite/internal/ducktest/tests/cellular_affinity_test/CellularTxStreamer.java:73]: each key has it's own map

Comment (ivandasch) [reply modules/ducktests/src/main/java/org/apache/ignite/internal/ducktest/tests/cellular_affinity_test/CellularTxStreamer.java:73]: Sorry, collections difers from each other on each iteration

File: modules/ducktests/src/main/java/org/apache/ignite/internal/ducktest/tests/pme_free_switch_test/SingleKeyTxStreamerApplication.java (hunk 3 above)
Comment (anton-vinogradov) [reply modules/ducktests/src/main/java/org/apache/ignite/internal/ducktest/tests/pme_free_switch_test/SingleKeyTxStreamerApplication.java]: Fixed.

File: modules/ducktests/src/main/java/org/apache/ignite/internal/ducktest/tests/cellular_affinity_test/CellularTxStreamer.java (hunk 2 above)
Comment (anton-vinogradov) [reply modules/ducktests/src/main/java/org/apache/ignite/internal/ducktest/tests/cellular_affinity_test/CellularTxStreamer.java]: Fixed.

File: modules/ducktests/src/main/java/org/apache/ignite/internal/ducktest/tests/cellular_affinity_test/CellularTxStreamer.java (hunk 1 above)
Comment (anton-vinogradov) [reply modules/ducktests/src/main/java/org/apache/ignite/internal/ducktest/tests/cellular_affinity_test/CellularTxStreamer.java]: Fixed.

**Review by ivandasch — APPROVED** [review]

**Pull request merged**